## Zad 1

In [2]:
import pandas as pd
import numpy as np
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


In [3]:
pensje = pd.Series({
    'Anna': 5500,
    'Jan': 7200,
    'Ewa': 4800,
    'Marek': 9100,
    'Kasia': 6300
})

print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


In [4]:
# 1. Kto zarabia najwięcej
print(pensje.idxmax())

# 2. Średnia pensja
print(pensje.mean())

# 3. Powyżej średniej
print(pensje[pensje > pensje.mean()])

# 4. Ile osób poniżej 6000
print((pensje < 6000).sum())

Marek
6580.0
Jan      7200
Marek    9100
dtype: int64
2


In [5]:
# 1. Podwyżka 10%
pensje_10 = pensje * 1.10

# 2. Premia 500
pensje_500 = pensje + 500

# 3. Nowa pensja
nowe_pensje = (pensje * 1.10) + 500

# 4. Ranking
print(nowe_pensje.sort_values(ascending=False))

Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


## Zad 2

In [8]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

In [9]:
print(sklep.shape)
print(sklep.columns)
print(sklep.dtypes)
print(sklep.head(3))

(5, 4)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


In [11]:
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']
sklep['cena_z_vat'] = sklep['cena'] * 1.23

print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [12]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

In [13]:
tips = pd.read_csv(url)

print(tips.shape)
print(tips.tail())
print(tips.sample(3))
print(tips.dtypes)

(244, 7)
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2
     total_bill   tip     sex smoker  day    time  size
4         24.59  3.61  Female     No  Sun  Dinner     4
223       15.98  3.00  Female     No  Fri   Lunch     3
75        10.51  1.25    Male     No  Sat  Dinner     2
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


## Zad 3

In [14]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

In [16]:
print(penguins.head(10))
print(penguins.shape)
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

In [17]:
print(penguins.describe())
print(penguins.describe(include='all'))
print(penguins['body_mass_g'].median())

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   

In [25]:
print(penguins['species'].value_counts())
print(penguins['island'].value_counts())
print(penguins['sex'].value_counts())
print(penguins['species'].value_counts(normalize=True))

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64
sex
MALE      168
FEMALE    165
Name: count, dtype: int64
species
Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674
Name: proportion, dtype: float64


In [26]:
print(penguins.isna().sum())

missing_sex = penguins['sex'].isna().sum()
percent = missing_sex / len(penguins) * 100
print(percent)

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64
3.1976744186046515


Dataset zawiera 344 pingwiny z 3 gatunków.
Najczęstszy gatunek to Adelie (152 osobniki, ~44.2%).
Mediana masy ciała wynosi 4050 g a średenia 4201 g
Dane zawierają braki w kilku kolumnach, np. sex (11 braków, 3.2%).
Pingwiny występują na 3 wyspach, z czego najwięcej na jednej dominującej (Biscoe).

## Zad 4

In [27]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

In [29]:

print(tips['total_bill'].mean())

print(tips['tip_pct'].mean())

print(tips['day'].value_counts())

print(tips['smoker'].value_counts())

print(tips['tip'].describe())

print(tips['time'].value_counts(normalize=True))

print(tips['size'].nunique())



19.78594262295082
16.08155737704918
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64
smoker
No     151
Yes     93
Name: count, dtype: int64
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
time
Dinner    0.721311
Lunch     0.278689
Name: proportion, dtype: float64
6
